lib/active_model/serializer/adapter/json.rb

```diff
@@ -47,6 +26,147 @@ def fragment_cache(cached_hash, non_cached_hash)
           Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
         end
 
+        private
+
+        # iterate through the associations on the serializer,
+        # adding them to the parent as needed (as singular or plural)
+        #
+        # nested_associations is a list of symbols that governs what
+        # associations on the passed in seralizer to include
+        def add_resource_relationships(parent, serializer, nested_associations = [])
+          # have the include array normalized
+          nested_associations = include_array_to_hash(nested_associations)
+
+          included_associations = if nested_associations.present?
```

Comment:
so `nested_associations` should be an normalized array with the nested objects you need.

Eg.
`BlogSerializer` with `has_many :posts include: [:author]`

so `nested_associations` will tell you that you need `author` but inside this if you check if it is inside `serializer.associations`, that will have only `posts`.

Initially I thought this was to check if it an already rendered relationship and avoid render it again. but it seems it is used [here](https://github.com/rails-api/active_model_serializers/pull/1114/files#diff-7b562c84aa219fb9dbfb8112b95df4d9R50) to loop the associations.

What I presume is the the code inside this if might never be executed unless you have something like (probably):

Eg.
`BlogSerializer` with `has_many :posts include: [:author.posts]`